gateways/authorizenet/authorizenet.go

```diff
@@ -134,6 +135,24 @@ func (client *AuthorizeNetClient) RefundWithContext(ctx context.Context, request
 		return nil, err
 	}
 
+	var transactionDetailsResponse *sleet.TransactionDetailsResponse
+	if request.Options != nil && request.Options[sleet.GooglePayTokenOption] != nil {
```

Comment:
wonder if it would be better to move these logic inside the `buildRefundRequest`? we can query `GetTransactionDetails` before calling `buildRefundRequest` if needed.

Reply:
No I think we want to separate this from buildRefundRequest